src/oop/programs_with_multiple_classes/medium/q8/MainTest.java

```diff
@@ -0,0 +1,137 @@
+            Object kitchenInstance = kitchen.createInstance(arguments);
+            _assertEquals(size, kitchen.getFieldValue(kitchenInstance, varSize),
+                    "Your " + classKitchen + " constructor does not correctly initialize the " + varSize + " field.");
+            _assertEquals(personCapacity, kitchen.getFieldValue(kitchenInstance, varCap),
+                    "Your " + classKitchen + " constructor does not correctly initialize the " + varCap + " field.");
+        }
+    }
+
+    @Test
+    public void kitchenClassHasCorrectAttributes() {
+        String incorrectFieldMessage = "Your " + classKitchen + " class is missing a required field.";
+        String incorrectModifierMessage = "One of your " + classKitchen + " class attributes does not have the correct modifier.";
+        assertTrue(kitchen.hasField(varSize, int.class), incorrectFieldMessage);
+        assertTrue(kitchen.hasModifier(varSize, "private"), incorrectModifierMessage);
+        assertTrue(kitchen.hasField(varCap, int.class), incorrectFieldMessage);
+        assertTrue(kitchen.hasModifier(varCap, "private"), incorrectModifierMessage);
+        assertTrue(kitchen.hasField(varLights, Light[].class), incorrectFieldMessage);
+        assertTrue(kitchen.hasModifier(varLights, "private"), incorrectModifierMessage);
+    }
+
+    @Test
+    public void lightClassHasRequiredConstructor() {
+        Class<?>[] classArguments = {boolean.class, String.class};
+        assertTrue(light.hasConstructor(classArguments),
+                "Your " + classLight + " constructor does not have the correct parameters.");
+        assertTrue(light.hasModifier(classArguments, "public"),
+                "Your " + classLight + " constructor does not have the correct modifier.");
+    }
+
+    private static Stream<Arguments> lightInputProvider() {
+        return Stream.of(
+                Arguments.of(true, "Plastic"),
+                Arguments.of(false, "Wood"),
+                Arguments.of(false, "Steel"),
+                Arguments.of(true, "Titanium")
+        );
+    }
+
+    @ParameterizedTest
+    @MethodSource("lightInputProvider")
+    public void lightConstructorInitializesValuesCorrectly(boolean isOn, String type) throws Throwable {
+        Object[][] arguments = {
+                {isOn, boolean.class},
+                {type, String.class}
+        };
+        Object lightInstance = light.createInstance(arguments);
+        _assertEquals(isOn, light.getFieldValue(lightInstance, varOn),
+                "Your " + classLight + " constructor does not correctly initialize the " + varOn + " field.");
+        _assertEquals(type, light.getFieldValue(lightInstance, varType),
+                "Your " + classLight + " constructor does not correctly initialize the " + varType + " field.");
+    }
+
+    private static Stream<Arguments> kitchenInputProvider() {
+        return Stream.of(
+                Arguments.of(-10, 15),
+                Arguments.of(-20, 30),
```

Comment:
These test cases should change. Some for each value should be positive and negative, and you should test the edge cases (0, 1, -1)

Reply:
In this case, the constructor is already given they should only check if it is negative and make it zero in that case.